src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const listOfStates = [];
+  const newState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(newState, action.extraData);
+        listOfStates.push({ ...newState });
+        break;
+      case 'removeProperties':
+        removeProperties(newState, action.keysToRemove);
+        listOfStates.push({ ...newState });
+        break;
+      case 'clear':
+        clear(newState);
+        listOfStates.push({});
+        break;
+    }
```

Comment:
The switch statement is missing a 'default' case for error handling. Always include a 'default' case to handle unexpected 'action.type' values.